src/content/developers/docs/consensus-mechanisms/pos/block-proposal/index.md

```diff
@@ -0,0 +1,71 @@
+---
+title: Block proposal
+description: Explanation of how blocks are proposed in proof-of-stake Ethereum.
+lang: en
+---
+
+Blocks are the fundamental units of the blockchain. This page explains how they are produced.
+
+## Prerequisites {#prerequisites}
+
+Block production is part of the proof-of-stake protocol. To help understand this page, we recommend you read about [proof-of-stake](src/content/developers/docs/consensus-mechanisms/pos/) and [block architecture](src/content/developers/docs/blocks/).
+
+## Who produces blocks? {#who-produces-blocks}
+
+Blocks are produced by validators. Validators are node operators that run validator software as well as their execution and consensus clients, and have deposited at least 32 ETH into the deposit contract. However, each individual validator is only occasionally responsible for producing a block. Time in Etheruem is broken into slots and epochs. Each slot is 12s in real-world time, and 32 slots (6.4 minutes) make up an epoch. Every slot is an opportunity for a block to be added to the blockchain.
+
+In Ethereum's proof-of-stake mechanism, a single validator is chosen to propose a block in each slot. This works by accumulating randomness from block to block using an algorithm called RANDAO. The block proposer adds a random value to the block that is mixed with the 'accumulated' RANDAO value when the block is processed. The random value gets its entropy from the validator's private key, which itself is generated sufficiently randomly, the total number of active validators, missed blocks, and the validator's effective balance.
```

Comment:
```suggestion
A single validator is pseudo-randomly chosen to propose a block in each slot. There is no such thing as true randomness in a blockchain because if each node generated truly random numbers they couldn't come to consensus. The aim is therefore to make the validator selection unpredictable. The way this is done in Ethereum is using an algorithm called RANDAO that 
mixes a hash from the block producer with a seed that gets updated every block. This value is used to select a specific validator from the total validator set. The validator selection is fixed four epochs in advance as a way to protect against certain kinds of seed manipulation.
```

Reply:
I like the suggested changes... (Would just remove the line break in there)

> mixes a hash from the block producer

Also, to my point above about clarifying between "producer" vs "proposer", would it be more appropriate here to say "mixed a hash from the block proposer"? I'm not actually sure when this randomness is injected... is it from whoever built the block, or from whoever proposes the block?